sources/ghostdog/src/infiniband.rs

```diff
@@ -0,0 +1,197 @@
+use crate::error;
+use crate::error::Result;
+use snafu::{ensure, ResultExt};
+use std::ffi::OsString;
+use std::ops::BitAnd;
+use std::path::Path;
+use std::str::FromStr;
+use std::{fs, str};
+
+const SYS_CLASS_DIR: &str = "/sys/class";
+const SW_MNG_CLASS: &str = "infiniband";
+const SW_MNG_VALUE: &str = "SW_MNG";
+
+/// Generic device struct found from sysfs
+#[derive(Clone, Debug)]
+pub(crate) struct SysDevice {
+    pub(crate) name: OsString,
+}
+
+/// Finds all the devices in sysfs for a given class sorted by name
+pub(crate) fn find_devices_by_class(class: String) -> Result<Vec<SysDevice>> {
```

Comment:
I would make this `find_infiniband_devices`, and have it return `Result<Vec<InfinibandDevice>>`, so we can punt on the problem of adding stronger typing than `String` to the class selector.

Later, if needed, we can make it more generic later by making class an enum, with variants like `Infiniband`, `Block` etc - but that will make the other functions and the return type more complex and it doesn't seem worth it for now.